serialization.md

```diff
@@ -468,62 +462,89 @@ Merkle Patricia Tree
                         | MerkleExtension ( ByteArray, MerkleTree )
                         | MerkleLeaf      ( ByteArray, String )
                         | ".MerkleTree"
-                        | ".MerkleBranch"            [function]
  // -----------------------------------------------------------
-    rule .MerkleBranch
-      => MerkleBranch (  0 |-> .MerkleTree  1 |-> .MerkleTree  2 |-> .MerkleTree  3 |-> .MerkleTree
-                         4 |-> .MerkleTree  5 |-> .MerkleTree  6 |-> .MerkleTree  7 |-> .MerkleTree
-                         8 |-> .MerkleTree  9 |-> .MerkleTree 10 |-> .MerkleTree 11 |-> .MerkleTree
-                        12 |-> .MerkleTree 13 |-> .MerkleTree 14 |-> .MerkleTree 15 |-> .MerkleTree
-                      , ""
-                      )
 
     syntax MerkleTree ::= MerkleUpdate ( MerkleTree,     String, String ) [function]
                         | MerkleUpdate ( MerkleTree,  ByteArray, String ) [function,klabel(MerkleUpdateAux)]
+                        | MerklePut    ( MerkleTree,  ByteArray, String ) [function]
+                        | MerkleDelete ( MerkleTree,  ByteArray )         [function]
  // --------------------------------------------------------------------------------------------------------
     rule MerkleUpdate ( TREE, S:String, VALUE ) => MerkleUpdate ( TREE, #nibbleize ( #parseByteStackRaw( S ) ), VALUE )
 
-    rule MerkleUpdate ( .MerkleTree, PATH:ByteArray, VALUE ) => MerkleLeaf ( PATH, VALUE )
+    rule MerkleUpdate ( TREE, PATH:ByteArray, VALUE ) => MerklePut ( TREE, PATH, VALUE ) requires VALUE =/=String ""
+    rule MerkleUpdate ( TREE, PATH:ByteArray, ""    ) => MerkleDelete ( TREE, PATH )
```

Comment:
Is this actually what should be happening? Writing an empty value means deleting? I thought the empty string shoudl be encoded as `\x80`?

Reply:
Yes. A merkle tree does not contain values that are empty strings. Both implementations I looked at had deleting from a tree done by updating the path with the empty string.